File: java/src/main/java/com/edulink/gui/models/courses/Resource.java
package com.edulink.gui.models.courses;

public class Resource {
    private int id;
    private int coursId;
    private int authorId;
    private String title;
    private String url;
    private String type;
    private String status;

    public Resource() {}

    public Resource(int id, int coursId, int authorId, String title, String url, String type, String status) {
        this.id = id;
        this.coursId = coursId;
        this.authorId = authorId;
        this.title = title;
        this.url = url;
        this.type = type;
        this.status = status;
    }

    public Resource(int coursId, int authorId, String title, String url, String type, String status) {
        this.coursId = coursId;
        this.authorId = authorId;
        this.title = title;
        this.url = url;
        this.type = type;
        this.status = status;
    }

    public int getId() { return id; }
    public void setId(int id) { this.id = id; }
    public int getCoursId() { return coursId; }
    public void setCoursId(int coursId) { this.coursId = coursId; }
    public int getAuthorId() { return authorId; }
    public void setAuthorId(int authorId) { this.authorId = authorId; }
    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }
    public String getUrl() { return url; }
    public void setUrl(String url) { this.url = url; }
    public String getType() { return type; }
    public void setType(String type) { this.type = type; }
    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }

    @Override
    public String toString() {
        return title + " (" + type + ")";
    }
}

File: ai/helprequest_models/classification_api/Entity/Resource.php
<?php

namespace App\Entity;

use App\Repository\ResourceRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;

#[ORM\Entity(repositoryClass: ResourceRepository::class)]
#[UniqueEntity(fields: ['title', 'cours'], message: 'This resource title already exists for this course.')]
class Resource
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 255, nullable: true)]
    #[Assert\NotBlank(message: "Title is required.")]
    #[Assert\Length(max: 255)]
    private ?string $title = null;

    #[ORM\Column(length: 255, nullable: true)]
    private ?string $url = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "Please select a resource type.")]
    private ?string $type = null;

    #[ORM\ManyToOne(inversedBy: 'resources')]
    #[ORM\JoinColumn(nullable: false, onDelete: "CASCADE")]
    private ?Cours $cours = null;

    #[ORM\Column(length: 20, options: ['default' => 'APPROVED'])]
    private ?string $status = 'APPROVED';

    #[ORM\ManyToOne(inversedBy: 'resources')]
    private ?User $author = null;

    public function getStatus(): ?string
    {
        return $this->status;
    }

    public function setStatus(string $status): static
    {
        $this->status = $status;

        return $this;
    }

    public function getCours(): ?Cours
    {
        return $this->cours;
    }

    public function setCours(?Cours $cours): static
    {
        $this->cours = $cours;

        return $this;
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getTitle(): ?string
    {
        return $this->title;
    }

    public function setTitle(?string $title): static
    {
        $this->title = $title;

        return $this;
    }

    public function getUrl(): ?string
    {
        return $this->url;
    }

    public function setUrl(?string $url): static
    {
        $this->url = $url;

        return $this;
    }

    /**
     * Returns the URL with a protocol prefix.
     * Prevents relative URL issues when rendering links in templates.
     */
    public function getAbsoluteUrl(): ?string
    {
        if ($this->url === null) {
            return null;
        }
        if (str_starts_with($this->url, 'http://') || str_starts_with($this->url, 'https://') || str_starts_with($this->url, '/')) {
            return $this->url;
        }
        return 'https://' . $this->url;
    }

    public function getType(): ?string
    {
        return $this->type;
    }

    public function setType(string $type): static
    {
        $this->type = $type;

        return $this;
    }

    public function getAuthor(): ?User
    {
        return $this->author;
    }

    public function setAuthor(?User $author): static
    {
        $this->author = $author;

        return $this;
    }
}
